packages/opensrp-reducer-factory/src/index.ts

```diff
@@ -0,0 +1,242 @@
+import { get, keyBy, values, Dictionary } from 'lodash';
+import { AnyAction, Store } from 'redux';
+import SeamlessImmutable from 'seamless-immutable';
+
+/** creates a union types that correctly matches against keys belonging to Obj whose value is of type keyType */
+export type KeysWhoseValuesAreType<Obj, KeyType> = {
+  [K in keyof Obj]: Obj[K] extends KeyType ? K : never;
+}[keyof Obj];
+export type ItemsIdFieldType<T> = KeysWhoseValuesAreType<T, string | number>;
+
+// actions
+
+/** FETCHED action type */
+export const FETCHED = 'opensrp/reducer/objects/FETCHED';
+/** REMOVE action type */
+export const REMOVE = 'opensrp/reducer/objects/REMOVE';
+/** SET_TOTAL_RECORDS action type */
+export const SET_TOTAL_RECORDS = 'opensrp/reducer/objects/SET_TOTAL_RECORDS';
```

Comment:
Is it possible to make these configurable (keeping these as the default) in the factory function?